plugin/src/main/java/com/denizenscript/denizen/objects/properties/entity/EntityAttributeModifiers.java

```diff
@@ -81,27 +84,64 @@ public static MapTag mapify(AttributeModifier modifier) {
         result.putObject("name", new ElementTag(modifier.getName()));
         result.putObject("amount", new ElementTag(modifier.getAmount()));
         result.putObject("operation", new ElementTag(modifier.getOperation()));
+        if (NMSHandler.getVersion().isAtLeast(NMSVersion.v1_20)) {
+            result.putObject("slot_group", new ElementTag(modifier.getSlotGroup().toString(), true));
+        }
+        if (NMSHandler.getVersion().isAtLeast(NMSVersion.v1_21)) {
+            result.putObject("key", new ElementTag(Utilities.namespacedKeyToString(modifier.getKey()), true));
+        }
+        // TODO: remove/deprecate these two
         result.putObject("slot", new ElementTag(modifier.getSlot() == null ? "any" : modifier.getSlot().name()));
-        if (NMSHandler.getVersion().isAtMost(NMSVersion.v1_20)) {
+        try {
             result.putObject("id", new ElementTag(modifier.getUniqueId().toString()));
         }
+        catch (Exception ignored) {}

```

Comment:
you just took my patch and made it significantly worse here for not clear reasons. try/catch is costly and problematic and should be avoided except for cases of genuine error handling. As a catch for "This will actually just not work 50% of the time" this is bad handling.

Reply:
If you want to add support for legacy id values to still be included, needs at least an `if (string.length() == 36 && string.indexOf('-') > 0)` check before trying to parse the `UUID.fromString` (see eg `PlayerTag.valueOf`)

Reply:
The `try/catch` is cause on older versions it'll just always work (which, correct me if I'm wrong, but I think if it just goes through without problems it's fine performance wise? can change that to a version check -> just add it if not though), but on newer versions it'll sometimes work for legacy custom attributes, and throw an exception otherwise, so a `try/catch` seemed like a good way to handle that - do you think we should just completely exclude the UUID on `1.21`+ though?

Reply:
Actually looks like Spigot now returns a V3 UUID for Attributes without an actual legacy UUID, so we can probably check this without a `try/catch`? either way the question (of removing it) still stands

Reply:
should leave it in at least for now, but... yeah either make sure it won't have exceptions, or add better prechecks to reduce the odds of an exception firing.

Broadly, in Java, any time any exception fires _something has gone wrong_, and Java is allowed to make a mess because it now has to spend time cleaning up a mess in your code. This is not ideal on live servers. And Java's backup plan for avoiding too much production impact from bad exception spam code, is it has a counter after which point it starts refusing to load exception info. This reduces the performance impact, but now makes debugging hell for anyone encountering a real issue.
So in short: it's much better to add extra checks to prevent an exception from firing, than to just handle the exception.

Reply:
Just leaves the UUID in now, since Spigot changed the method to always provide a UUID